sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/ContinuousReader.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.spark.sql.sources.v2.reader;
+
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow;
+import org.apache.spark.sql.execution.streaming.Offset;
+import org.apache.spark.sql.sources.v2.BaseStreamingSource;
+import org.apache.spark.sql.types.StructType;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A mix-in interface for {@link DataSourceV2Reader}. Data source readers can implement this
+ * interface to allow reading in a continuous processing mode stream.
+ *
+ * Implementations must ensure each read task output is a {@link ContinuousDataReader}.
+ */
+public interface ContinuousReader extends BaseStreamingSource, DataSourceV2Reader {
+    /**
+     * Merge offsets coming from {@link ContinuousDataReader} instances in each partition to
+     * a single global offset.
+     */
+    Offset mergeOffsets(Offset[] offsets);
```

Comment:
I feel this method should be in Offsets rather than in ContinuousReader.
Also, having `Offset` represent both merged offset to be saved in the log as well as the offset for each partition is often confusing. Looking at the API its not clear which method deals with merged offsets and partial offsets.

Reply:
I agree that it'd make sense to add some kind of PartitionOffset interface. It'd be pretty lightweight, since I think we can just insist it be natively serializable.

Is there a way to let execution get a handle to the right mergeOffsets if it's not in the reader?